relay-server/src/services/project_cache.rs

```diff
@@ -827,38 +774,71 @@ impl ProjectCacheBroker {
     ///
     /// The flushing of the buffered envelopes happens in `update_state`.
     fn handle_validate_envelope(&mut self, message: ValidateEnvelope) {
-        let ValidateEnvelope { envelope: context } = message;
+        // TODO: check_envelope here?
+
+        let ValidateEnvelope {
```

Comment:
All the project checks are now done in `handle_validate_envelope`. There is no more redundant check in `handle_processing`.